Source/Search/Mixins/AdvancedSearchProviderMixin.lua

```diff
@@ -0,0 +1,143 @@
+local _, addonTable = ...
+local zc = addonTable.zc
+
+AuctionatorAdvancedSearchProviderMixin = CreateFromMixins(AuctionatorMultiSearchMixin, AuctionatorSearchProviderMixin)
+
+local KEYWORD_SEARCH_EVENTS = {
+  "AUCTION_HOUSE_BROWSE_RESULTS_UPDATED",
+  "AUCTION_HOUSE_BROWSE_RESULTS_ADDED",
+  "AUCTION_HOUSE_BROWSE_FAILURE"
+}
+
+local function IsCompoundSearch (searchString)
+  if searchString == nil then
+    return false;
+  else
+    return zc.StringContains (searchString, ">") or
+      zc.StringContains (searchString, Auctionator.Constants.AdvancedSearchDivider);
+  end
+end
+
+local function ParseCompoundSearch( searchString )
+  local delimiter = Auctionator.Constants.AdvancedSearchDivider
+
+  if zc.StringContains( searchString, ">" ) then
+    delimiter = ">"
+  end
+
+  local queryString, filterKey, minLevel, maxLevel, minItemLevel, maxItemLevel =
+    strsplit( delimiter, searchString )
+  --TODO Convert filterKey to filters/itemClassFilters in the browse query
+
+  --TODO Check for searching the whole AH by mistake (which causes a DC)
```

Comment:
Let's just `error` out if the `queryString` is nil (do we need to check for empty string too?):
```lua
if queryString == nil then
  error("A search term is required to search the AH with Auctionator.")
end
```

Reply:
This was